src/state/routing/slice.ts

```diff
@@ -15,9 +18,29 @@ const DEFAULT_QUERY_PARAMS = {
   protocols: protocols.map((p) => p.toLowerCase()).join(','),
 }
 
+const serializeRoutingCacheKey: SerializeQueryArgs<any> = ({ endpointName, queryArgs }) => {
+  // same as default serializeQueryArgs, but we add extra case if key is our non-serializable JsonRpcProvider
+  return `${endpointName}(${JSON.stringify(queryArgs, (key, value) => {
```

Comment:
I don't think you still need to serialize this, now that you've disabled serialization checks.